### PR TITLE
Change OpenMod.EntityFrameworkCore to use Pomelo.EntityFrameworkCore.MySql

### PR DESCRIPTION
**Pull request by IAmSilK**
Todo:
- [ ] Update UserDatabasePlugin to reflect changes and create new migrations
- [x] Test new implementations work
- [ ] Changed OpenMod docs to reflect changes

### CODE REVIEW

**Comment by IAmSilK** [comment]
#524

File: framework/OpenMod.EntityFrameworkCore.MySql/OpenModMySqlDbContext.cs [hunk 1]
```diff
@@ -1,60 +1,35 @@
...
+namespace OpenMod.EntityFrameworkCore.MySql
 {
     /// <summary>
     /// Base <see cref="DbContext"/> for OpenMod plugins.
     /// </summary>
     /// <typeparam name="TSelf">A type parameter pointing to the class implementing itself.</typeparam>
     /// <example>
     /// <code>
-    /// public class MyDbContext : OpenModDbContext&lt;MyDbContext&gt;
+    /// public class MyDbContext : OpenModMySqlDbContext&lt;MyDbContext&gt;
```
Comment (Trojaner) [inline framework/OpenMod.EntityFrameworkCore.MySql/OpenModMySqlDbContext.cs:14]: Should stay as OpenModDbContext

File: framework/OpenMod.EntityFrameworkCore.MySql/OpenModMySqlDbContext.cs [hunk 2]
```diff
@@ -74,23 +49,11 @@ protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
             var connectionStringAccessor = m_ServiceProvider.GetRequiredService<IConnectionStringAccessor>();
             var connectionString = connectionStringAccessor.GetConnectionString(connectionStringName);
 
-            optionsBuilder.UseMySQL(connectionString,
+            optionsBuilder.UseMySql(connectionString,
```
Comment (Trojaner) [inline framework/OpenMod.EntityFrameworkCore.MySql/OpenModMySqlDbContext.cs:52]: This should be removed and done in a different way, e.g. in container/service configuration when you add the DbContext. DbContext itself should not care what the underlying provider is

**Review by Trojaner — CHANGES_REQUESTED** [review]

**Comment by Trojaner** [comment]
We can skip the UserDatabasePlugin for now, its not priority

**Review by Trojaner — APPROVED** [review]

**Comment by IAmSilK** [comment]
Testing is finished

**Pull request merged**